integration-provisioning/src/main/java/org/symphonyoss/integration/provisioning/client/AppRepositoryClient.java

```diff
@@ -168,4 +243,48 @@ public void updateApp(AppStoreWrapper appStoreApp, String userId, String appId)
     return headers;
   }
 
+  /**
+   * Get the required headers to be used by the HTTP requests.
+   * @param userId User identifier
+   * @return Required headers
+   */
+  private Map<String, String> getRequiredHeader(String userId) {
+    AuthenticationToken token = authenticationProxy.getToken(userId);
+    String sessionToken = token.getSessionToken();
+
+    Map<String, String> headers = new HashMap<>();
+    headers.put(SESSION_TOKEN_HEADER, sessionToken);
+
+    return headers;
+  }
+
+  /**
+   * Parse a AppStoreWrapper into a AdminApplicationWrapper
+   * @param appStoreWrapper {@link AppStoreWrapper } to be parsed.
+   * @return Parsed {@link AdminApplicationWrapper}.
+   */
+  private AdminApplicationWrapper parseWrapper(AppStoreWrapper appStoreWrapper) {
+    AdminApplicationDetailWrapper adminDetailWrapper = new AdminApplicationDetailWrapper();
+    adminDetailWrapper.setAppId(appStoreWrapper.getAppGroupId());
+    adminDetailWrapper.setDomain(appStoreWrapper.getDomain());
+    adminDetailWrapper.setName(appStoreWrapper.getName());
+    adminDetailWrapper.setPublisher(appStoreWrapper.getPublisher());
+
+    AdminApplicationWrapper adminWrapper = new AdminApplicationWrapper();
+    // Description length cannot exceed 300 characters (otherwise will be truncated)
+    String description = appStoreWrapper.getDescription();
```

Comment:
Good catch :+1: 

But, we should review all descriptions to avoid incomprehensive description when it was truncated.

Reply:
Task created https://perzoinc.atlassian.net/browse/APP-1638